src/components/BaseSeparatePage/components/Functions/index.module.scss

```diff
@@ -0,0 +1,143 @@
+@import 'src/styles/presets.module';
+@import '../../../../styles/variables.module';
+
+.functions {
+  display: flex;
+  justify-content: space-between;
+  font-weight: 400;
+
+  .functionsItemGroup {
+    :last-child {
+      border: none;
+    }
+  }
+
+  .functionsProgress {
+    min-width: 210px;
+    margin-top: 120px;
+
+    svg {
+      width: 19px;
+      height: 19px;
+    }
+
+    .onThisPage {
+      margin: 20px 0 25px;
+      font-weight: 400;
+      font-size: 16px;
+      line-height: 135%;
+    }
+
+    .progressBar {
+      width: 195px;
+      height: 220px;
+    }
+  }
+
+  .editPageButtonForMobile,
+  .onThisPageForMobile {
+    display: none;
+  }
+
+  @media (max-width: $mobileBreakPoint) {
+    display: block;
+    width: 100%;
+
+    .functionsProgress {
+      display: none;
+    }
+
+    .editPageButtonForMobile,
+    .onThisPageForMobile {
+      display: block;
+      width: 100%;
+      padding: 12px 24px;
+    }
+
+    .onThisPageForMobile {
+      width: 100%;
+
+      span {
+        display: flex;
+        align-items: center;
+        justify-content: space-between;
+        width: 100%;
+        height: 30px;
+        padding: 0 12px;
+        font-weight: 500;
+        font-size: 12px;
+        line-height: 105%;
+        letter-spacing: 0.05em;
+        border: 1px solid;
+        border-radius: 3px;
+
+        svg {
+          width: 7px;
+          height: 7px;
+          stroke: var(--colorPrimary);
+        }
+      }
+    }
+  }
+}
+
+.editPageButton {
+  display: flex;
+  flex-direction: row;
+  gap: 8px;
+  align-items: center;
+  height: 30px;
+  padding: 0 10px;
+  font-weight: 500;
+  font-size: 12px;
+  line-height: 105%;
+  letter-spacing: 0.11em;
+  background-color: #fff;
+  border: 1px solid #000;
+  border-radius: 3px;
+
+  .editPageString {
+    svg {
+      width: unset;
+      height: unset;
+    }
+
+    &::after {
+      display: inline-block;
+      width: 7px;
+      height: 7px;
+      background-image: url('../../../../styles/icons/arrow.svg');
```

Comment:
在 css 中这样引用算是隐式的耦合了，我建议任何的耦合都应该尝试让它们显式化，包括对资源的引用。
另外一点是尽量只引用同级的资源，如果多个组件要复用一个资源的话，可以将那个资源封装成一个组件。
这种设计方式主要是考虑到目前 ide 对于 ts、js 的代码引用追踪能力较强，对于 css、资源等方面的追踪能力不够强，所以维护、移除、重构时不一定能追踪到完整的引用链，就不一定能正确处理好代码，比如移除组件时少删了一部分资源之类的。

这里你要复用的话可以考虑封装成一个通用的 ExternalLink 组件，在 ts 中引入资源 url 或直接引入 svg 组件。

Reply:
assets could be placed in `public` directory and it will be resolved by next.js automatically. As a image at `/public/images/sample1.jpg` can be referenced by `/images/sample1.jpg`

Reply:
> 在 css 中这样引用算是隐式的耦合了，我建议任何的耦合都应该尝试让它们显式化，包括对资源的引用。 另外一点是尽量只引用同级的资源，如果多个组件要复用一个资源的话，可以将那个资源封装成一个组件。 这种设计方式主要是考虑到目前 ide 对于 ts、js 的代码引用追踪能力较强，对于 css、资源等方面的追踪能力不够强，所以维护、移除、重构时不一定能追踪到完整的引用链，就不一定能正确处理好代码，比如移除组件时少删了一部分资源之类的。
> 
> 这里你要复用的话可以考虑封装成一个通用的 ExternalLink 组件，在 ts 中引入资源 url 或直接引入 svg 组件。

嗯。 我可能会做成一个公共的class，进行引用。

Reply:
Components have been added to replace the original style.